CONTRIBUTING.md

```diff
@@ -18,18 +21,202 @@ jupyter lab
 
 Step 4) experiment within notebooks with what you wnat the end data scientist experience to be and then merge code in to the codebase as you go.
 
-### Git and Github
+### Learn Git and Github
 
-All our development is done using Git and Github. If you're not too familiar with Git and Github, [start by reviewing this guide](https://guides.github.com/activities/hello-world).
+All our development is done using Git and Github. If you're not too familiar with Git and Github, start by [reviewing this guide](https://guides.github.com/activities/hello-world).
 
 ### Slack
 
 [Join our Slack community](http://slack.openmined.org).
 
+### Issues
+
+On <https://github.com/OpenMined/PySyft/issues> you can find all open Issues. You can find a detailed explanation on how to work with issues below under [Issue Allocation](#issue-allocation).
+
+## Setup
+
+### Forking a Repository
+
+To contribute to PySyft you will need to fork the OpenMind/PySyft repository.
+Then you can work risk-free on your fork.
+
+Fork this Repository. This will create a local copy (Remote Repository) of this repository to your Github profile. Keep a reference to the original project in `upstream` remote.
+```
+$ git clone https://github.com/<your-username>/<repo-name>
+$ cd <repo-name>
+$ git remote add upstream https://github.com/<upstream-owner>/<repo-name>
+```
+
+### Syncing a Forked Repository
+
+To sync your fork (remote) with the OpenMined/PySyft (upstream) repository please see this [Guide](https://help.github.com/articles/syncing-a-fork/) on how to sync your fork or follow the given commands.
+
+```
+$ git remote update
+$ git checkout <branch-name>
+$ git rebase upstream/<branch-name>
+```
+
 ### Setting up Pre-Commit Hook
 
 PySyft uses the python package `pre-commit` to make sure the correct formatting (black & flake) is applied.
 
 You can install it via `pip install pre-commit`
 
 Then you just need to call `pre-commit install`
+
+This can all also be done by running `make install_hooks`
+
+### Syncing a Forked Repository
```

Comment:
I think this part is duplicated

Reply:
Good spot.

Reply:
Addressed here: https://github.com/OpenMined/PySyft/pull/4746